xbmc/platform/darwin/osx/network/NetworkOsx.cpp

```diff
@@ -37,22 +37,37 @@ std::string CNetworkInterfaceOsx::GetCurrentDefaultGateway() const
   std::string result;
 
   FILE* pipe = popen("echo \"show State:/Network/Global/IPv4\" | scutil | grep Router", "r");
-  usleep(100000);
+
   if (pipe)
   {
     std::string tmpStr;
     char buffer[256] = {'\0'};
-    if (fread(buffer, sizeof(char), sizeof(buffer), pipe) > 0 && !ferror(pipe))
+    struct pollfd fds[1];
+    fds[0].fd = fileno(pipe);
+    fds[0].events = POLLIN;
+
+    int pollResult = poll(fds, 1, 100); // Wait for 100 milliseconds
+    if (pollResult > 0 && (fds[0].revents & POLLIN))
     {
-      tmpStr = buffer;
-      if (tmpStr.length() >= 11)
-        result = tmpStr.substr(11);
+      const int lineLength = 11;
+      ssize_t bytesRead = fread(buffer, sizeof(char), sizeof(buffer) - 1, pipe);
```

Comment:
Can you not read directly into a std::string rather than using a temporary buffer? Use std::string::resize before and after.

Reply:
done